File: jscomp/test/shift_test.js
'use strict';


function to_unsgined(x) {
  return (x >>> 0);
}

function f(x) {
  return (x >>> 0);
}

function ff(x) {
  return (x >>> 0);
}

function fff(x) {
  return 3 + (3 + (4 + (1 + x)));
}

exports.to_unsgined = to_unsgined;
exports.f = f;
exports.ff = ff;
exports.fff = fff;
/* No side effect */

File: jscomp/test/shift_test.ml
let to_unsgined x = Nativeint.shift_right_logical x 0
let f (x : nativeint) = to_unsgined (to_unsgined (to_unsgined x))

(* TODO: this should be optimized as a single [to_unsigned]*)
let ff (x : nativeint) = x |> to_unsgined |> to_unsgined

open Nativeint

let fff x = add 3n @@ add 3n @@ add 4n @@ add 1n x
